index.js

```diff
@@ -145,11 +146,32 @@ function handleError(err) {
   console.error(err);
 }
 
+function getNextTimeout(){
+	var now = new Date();
+	var waitingTime;
+	startTime.setHours(22);
+	startTime.setMinutes(0);
+	waitingTime = startTime.getTime() - now.getTime();
+	console.log('Next search starts in '  + waitingTime/(60*1000) + ' minutes.');
+	return waitingTime;
+}
+
 (function initialize() {
-  var searchInterval = setInterval(search, LIMIT_TIME / LIMIT_GET);
-  var publishInterval = setInterval(publish, LIMIT_TIME / LIMIT_POST);
-  setTimeout(function () {
-    clearInterval(searchInterval);
-    clearInterval(publishInterval);
-  }, LIMIT_TIME);
+	var nextEjecTime = getNextTimeout();
+	if(nextEjecTime < 0){
+		startTime.setDate(startTime.getDate + 1);
+		initialize();
+	}
+	else{
+		setTimeout(function(){
+			var searchInterval = setInterval(search, LIMIT_TIME / LIMIT_GET);
+			var publishInterval = setInterval(publish, LIMIT_TIME / LIMIT_POST);
+			setTimeout(function () {
+			  clearInterval(searchInterval);
+			  clearInterval(publishInterval);
+			}, LIMIT_TIME);
+			startTime.setDate(startTime.getDate + 1);
+			initialize();
```

Comment:
I would move this inside the clearing function so that fewer things are "running" in background.